### PR TITLE
Fix a bug that RealmList.delete*() does not delete target objects

### LINKED ISSUE #5233: RealmList.delete*() does not delete target object.
`RealmList.delete*()` must delete target objects in addition to removing them from the list.
Due to a bug introduced our refactoring introduced in `3.7.1`, `RealmList.delete*()` (except for `deleteAllFromRealm()`) only remove the target objects from the list and target objects still remain in the database.

Following test exposes the bug.

```
@Test
public void deleteFromRealm() {
    Owner owner = realm.where(Owner.class).findFirst();
    //noinspection ConstantConditions
    RealmList<Dog> dogs = owner.getDogs();
    assertEquals(TEST_SIZE, dogs.size());

    int expectedSize = TEST_SIZE;
    realm.beginTransaction();
    dogs.deleteFromRealm(TEST_SIZE / 2);
    expectedSize--;
    realm.commitTransaction();
    assertEquals(expectedSize, dogs.size());
    assertEquals(expectedSize, realm.where(Dog.class).count());

    realm.beginTransaction();
    dogs.deleteFromRealm(0);
    expectedSize--;
    realm.commitTransaction();
    assertEquals(expectedSize, dogs.size());
    assertEquals(expectedSize, realm.where(Dog.class).count());

    realm.beginTransaction();
    dogs.deleteFromRealm(dogs.size() - 1);
    expectedSize--;
    realm.commitTransaction();
    assertEquals(expectedSize, dogs.size());
    assertEquals(expectedSize, realm.where(Dog.class).count());
}

@Test
public void deleteFirstFromRealm() {
    Owner owner = realm.where(Owner.class).findFirst();
    //noinspection ConstantConditions
    RealmList<Dog> dogs = owner.getDogs();
    assertEquals(TEST_SIZE, dogs.size());

    realm.beginTransaction();
    dogs.deleteFirstFromRealm();
    realm.commitTransaction();
    assertEquals(TEST_SIZE - 1, dogs.size());
    assertEquals(TEST_SIZE - 1, realm.where(Dog.class).count());
}

@Test
public void deleteLastFromRealm() {
    Owner owner = realm.where(Owner.class).findFirst();
    //noinspection ConstantConditions
    RealmList<Dog> dogs = owner.getDogs();
    assertEquals(TEST_SIZE, dogs.size());

    realm.beginTransaction();
    dogs.deleteLastFromRealm();
    realm.commitTransaction();
    assertEquals(TEST_SIZE - 1, dogs.size());
    assertEquals(TEST_SIZE - 1, realm.where(Dog.class).count());
}
```

And the result is:

```
io.realm.RealmListTests > deleteLastFromRealm[Nexus_5X_API_21(AVD) - 5.0.2] FAILED
        java.lang.AssertionError: expected:<9> but was:<10>
        at org.junit.Assert.fail(Assert.java:88)

io.realm.RealmListTests > deleteFirstFromRealm[Nexus_5X_API_21(AVD) - 5.0.2] FAILED
        java.lang.AssertionError: expected:<9> but was:<10>
        at org.junit.Assert.fail(Assert.java:88)

io.realm.RealmListTests > deleteFromRealm[Nexus_5X_API_21(AVD) - 5.0.2] FAILED
        java.lang.AssertionError: expected:<9> but was:<10>
        at org.junit.Assert.fail(Assert.java:88)
```

### PR DESCRIPTION
**Pull request by zaki50**
Fixes #5233

### CODE REVIEW

**Comment by dalinaum** [comment]
Shoud https://github.com/realm/realm-object-store/pull/540 be merged first?

File: realm/realm-library/src/androidTest/java/io/realm/RealmListTests.java [hunk 1]
```diff
@@ -763,8 +779,67 @@ public void realmMethods_onDeletedLinkView() {
     }
 
     @Test
-    public void removeAllFromRealm() {
+    public void deleteFromRealm() {
```
Comment (beeender) [inline realm/realm-library/src/androidTest/java/io/realm/RealmListTests.java]: we just need add assertion to `ManagedOrderedRealmCollectionTests`. I am fixing it now

**Comment by zaki50** [comment]
@dalinaum No need to merge that for now. On the merge of #5229 , we need it for `master` branch.

**Comment by zaki50** [comment]
@beeender thanks. I'll merge this.

**Pull request merged**